tools/Microsoft.Health.Extensions.BuildTimeCodeGenerator/Sql/CreateTableVisitor.cs

```diff
@@ -59,6 +39,125 @@ public override void Visit(CreateTableStatement node)
+
+            List<(string name, string alias)> tables =
+                querySpecification.FromClause.TableReferences
+                    .SelectMany(tr => tr switch
+                    {
+                        JoinTableReference @join => new[] { @join.FirstTableReference, @join.SecondTableReference },
+                        _ => new[] { tr },
+                    })
+                    .Select(tr => tr switch
+                    {
+                        NamedTableReference ntr => (ntr.SchemaObject.BaseIdentifier.Value, ntr.Alias.Value),
+                        _ => throw new NotSupportedException($"Unrecognized table type '{tr.GetType().Name}' in view.")
+                    }).ToList();
+
+            ClassDeclarationSyntax classDeclarationSyntax =
+                CreateSkeletalClass(className, schemaQualifiedViewName)
+                    .AddMembers(selectElements.Select(selectElement => CreatePropertyForViewColumn(selectElement, tables)).ToArray());
+
+            FieldDeclarationSyntax field = CreateStaticFieldForClass(className, viewName);
+
+            MembersToAdd.Add(field.AddSortingKey(this, viewName));
+            MembersToAdd.Add(classDeclarationSyntax.AddSortingKey(this, viewName));
+
+            base.Visit(node);
+        }
+
+        private MemberDeclarationSyntax CreatePropertyForViewColumn(SelectElement selectElement, List<(string name, string alias)> tablesInScope)
+        {
+            if (selectElement is not SelectScalarExpression exp)
+            {
+                // notably SELECT * is not supported
+                throw new NotSupportedException($"Select element {selectElement.GetType().Name} is not supported");
+            }
+
+            if (exp.Expression is not ColumnReferenceExpression columnReference)
+            {
+                throw new NotSupportedException($"{exp.Expression.GetType().Name} is not supported.");
+            }
+
+            if (columnReference.MultiPartIdentifier.Count != 2)
+            {
+                throw new NotSupportedException("Please qualify column references with the table's alias");
+            }
+
+            string tableAliasName = columnReference.MultiPartIdentifier[0].Value;
+            string tableColumnName = columnReference.MultiPartIdentifier[1].Value;
+
+            string tableName = tablesInScope.Where(t => t.alias == tableAliasName).Select(t => t.name).FirstOrDefault() ?? throw new InvalidOperationException($"Unable to resolve table alias '{tableAliasName}'.");
+
+            string classNameForTable = GetClassNameForTable(tableName);
+
+            // find the class we generated for the table
+            var tableDeclaration = (ClassDeclarationSyntax)MembersToAdd.FirstOrDefault(m => m is ClassDeclarationSyntax c && c.Identifier.ValueText == classNameForTable)
+                                   ?? throw new InvalidOperationException($"table not found");
```

Comment:
Nit: Perhaps specify which table

```suggestion
                                   ?? throw new InvalidOperationException($"Table '{classNameForTable}' was not found");
```